core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -73,6 +73,10 @@ class Automation {
             val militaryUnits = civUnits.filter { !it.type.isCivilian()}.size
             val workers = civUnits.filter { it.name == CityConstructions.Worker }.size
             val cities = cityInfo.civInfo.cities.size
+            val canBuildWorkboat = cityInfo.cityConstructions.getConstructableUnits().map { it.name }.contains("Work Boats")
+                    && !cityInfo.getTiles().any { val civUnit = it.civilianUnit; civUnit != null && civUnit!!.name == "Work Boats" }
```

Comment:
cityInfo.getTiles().any{it.civilianUnit?.name=="Work Boats"}

See https://kotlinlang.org/docs/reference/null-safety.html#safe-calls

Reply:
No prob, working on new languages is always a learning experience =)